Library/Context/ContextDto.php

```diff
@@ -144,4 +162,34 @@ public function getWorkflowStepListByPosition(int $positionNo) : array
         $this->di->get('logr')->debug(json_encode(func_get_args()));
         return $this->rawHash['steps'][$positionNo] ?? [];
     }
+
+    /**
+     * Returns the step node related to the given code
+     *
+     * @param string $code the step code
+     *
+     * @return array the step node
+     *               [
+     *                  'stepName' => ['foo' => 'bar'],
+     *               ]
+     */
+    public function getStep(string $code) : array
+    {
+        $this->di->get('logr')->debug(json_encode(func_get_args()));
+        foreach ($this->rawHash['steps'] as $stepHash) {
+            // if // steps
+            if (!(array_keys($stepHash) !== array_keys(array_keys($stepHash)))) {
```

Comment:
only one if dimension is not possible ? return $stepHash; first

Reply:
don't think so, because the step position or path is not predictable we have to run through all steps items to check the code.